Milestone #1

simandebvu opened the pull request:
## Added 

- HTML sizes for sm, md and lg screen types
- Relevant padding and margin
- Anchor tags for elements in page
- Screenshots for elements
- HTML and CSS Validation (Stickler)
- Site Screenshot encompassing device types

simandebvu committed:
Added HTML initial structure

simandebvu committed:
Styled the featured articles

simandebvu committed:
Finished Styling NavBar

simandebvu committed:
Resolved Nav Scrolling and Row Display

simandebvu committed:
Merge pull request #1 from simandebvu/feature-html

Feature html

simandebvu committed:
Added Debate Optiondivs

simandebvu committed:
Added Footer

simandebvu committed:
Added layout for md and lg sizes

simandebvu committed:
Styled navbar font and hover

simandebvu committed:
Added layout for md, fixed HTML validation

simandebvu committed:
Fixed Stylelint issues

simandebvu committed:
Merge pull request #2 from simandebvu/feature-html

Feature html

simandebvu committed:
Updated Nav Bar Links

simandebvu committed:
Added links

simandebvu committed:
Resolved Heroes Link

simandebvu committed:
Added aside links

simandebvu committed:
Updated Screenshot

simandebvu committed:
Merge pull request #4 from simandebvu/feature-html

Feature html

simandebvu committed:
Updated readme description

simandebvu committed:
Merge pull request #5 from simandebvu/feature-html

Milestone #1

rammazzoti2000 commented:
## Pull Request under review by Alexandru Bangau (TSE)

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+
+.navbar-brand img {
+  height: 20px;
+  padding-left: 3px;
```
`padding-left` css rule can be used as bootstrap class --> `pl-2`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+
+.nav-button {
+  font-size: 14px;
+  text-transform: uppercase;
```
`text-transform` css rule can be used as bootstrap class --> `text-uppercase` (simple as that :smiley: )

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+}
+
+.navbar-light .navbar-nav .nav-link {
+  color: rgba(0, 0, 0);
```
`color` (black) css rule can be used as a bootstrap class --> `text-dark`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+}
+
+.secondary-nav {
+  background-color: #fff !important;
```
no use of `!important` properties!
`background-color` (white) class selector can be used as a bootstrap class --> `bg-white`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+.nav-item a {
+  font-size: 12px;
+  line-height: 20px;
+  font-weight: bold;
```
`font-weight` rule css can be used as bootstrap class --> `font-weight-bold`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+  line-height: 20px;
+  font-weight: bold;
+  font-family: "Roboto Condensed", sans-serif;
+  text-transform: capitalize;
```
`text-transform`  css rule can be used as bootstrap class --> `text-capitalize`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+  color: #fff;
+  text-decoration: none;
+  background-color: transparent;
+  margin-right: 2px;
```
all these css rules can be used as bootstrap classes:
`text-white text-decoration-none bg-transparent mr-1`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+
+.sub-button {
+  color: var(--main-page-color);
+  background-color: #fff;
```
`background-color` css rule can be used as bootstrap class --> `bg-white`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+.sub-button {
+  color: var(--main-page-color);
+  background-color: #fff;
+  border-radius: 0;
```
`border-radius` css rule can be used as bootstrap class --> `rounded-0`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+.link-deco,
+.link-no-deco,
+.article-link {
+  color: #000;
```
`color` (black) css rule can be used as bootstrap class --> `text-dark`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+}
+
+.bold-text {
+  font-weight: bold;
```
font-weight rule css can be used as bootstrap class --> font-weight-bold

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+}
+
+p {
+  padding-top: 10px;
```
`padding-top` css rule can be used as bootstrap class --> `pt-3`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+}
+
+.opinion-item {
+  display: flex;
+  align-items: center;
+  padding: 5px;
```
all these css rules can be used as bootstrap classes:
`d-flex align-items-center p-2`

simandebvu replied:
Thank you so so much for this! Am much much wiser now 😄

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+  padding-top: 10px;
+  padding-bottom: 4px;
+  display: flex;
+  align-items: center;
```
same as above

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+
+.magazine-images {
+  position: relative;
+  text-align: center;
```
these css rules can be used as bootstrap classes
`position-relative text-center`

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,379 @@
+  font-weight: bold;
+  position: relative;
+  display: flex;
+  flex-direction: column;
```
these classes can be used as bootstrap classes
`d-flex flex-column position-relative font-weight-bold`

rammazzoti2000 reviewed (changes requested):
### Hello @simandebvu I'm Alex your code reviewer for this project :raised_hands: 
Good attempt this far on this project, I see a descriptive Readme file, Pull Request summary descriptive as well, the commits could've been made more often though (remember small commits with meaningful description), also good use on bootstrap classes but I think you can do better than this :muscle: 
So as you expected already, there are still issues to fix for this project before it cam be approved, follow the comments bellow :point_down: 
## :stop_sign: Required Changes :construction: :construction_worker: 
- [x] More use of bootstrap classes, remember this is a bootstrap project, see the inline comments :point_down: 
- [x] Overall the page needs more work on typography, colors and alignment
###
### On Large Screens (Desktop) :desktop_computer: 
- [x] The navigation bar is not as the original and the logo is too big, also the links are a bit off, should be center aligned, and you are also missing the weather widget
### Clone:
![image](https://user-images.githubusercontent.com/3244654/82722305-c2dd6100-9cc5-11ea-8e80-c02017ab6bc7.png)
### Original:
![image](https://user-images.githubusercontent.com/3244654/82722314-df799900-9cc5-11ea-984e-ca442a06ae66.png)
####
- [x] Overall the page is missing some space on left and right (meaning is taking all the width when it should not)
### Clone:
![image](https://user-images.githubusercontent.com/3244654/82722360-531ba600-9cc6-11ea-8be1-7c1c4d121874.png)
### Original:
![image](https://user-images.githubusercontent.com/3244654/82722366-6169c200-9cc6-11ea-9d6b-e44c172c0974.png)
 - [x] In the `Editor's Pick` section you are missing a row with four elements, the original has two rows with four elemenst each
![image](https://user-images.githubusercontent.com/3244654/82722407-c2919580-9cc6-11ea-80ed-4918f481cd0c.png)
- [x] Your `Subscribe` section is a bit off, it should look like the original
### Clone:
![image](https://user-images.githubusercontent.com/3244654/82722439-04bad700-9cc7-11ea-9dba-3180c85e4bc1.png)
### Original:
![image](https://user-images.githubusercontent.com/3244654/82722443-1603e380-9cc7-11ea-9e6f-d9802a96e664.png)
- [x] The footer is not like original, please check the screenshots
### Clone:
![image](https://user-images.githubusercontent.com/3244654/82722456-3b90ed00-9cc7-11ea-99e1-ab70ecc95fe2.png)
### Original:
![image](https://user-images.githubusercontent.com/3244654/82722461-4cd9f980-9cc7-11ea-81dd-3c4fd4169d59.png)
### On Medium Screens (Tablet) 
- [x] The `Featured Stories` section is missing some content, please add at least 3 more articles to it
- [x] The `Sign Up` feature should be like the original
### Clone:
![image](https://user-images.githubusercontent.com/3244654/82722626-ba3a5a00-9cc8-11ea-8b2d-46d3907660d0.png)

### Original:
![image](https://user-images.githubusercontent.com/3244654/82722616-a5f65d00-9cc8-11ea-907b-ae871804f2a9.png)
- [x] The `U.S., Sport, Business, etc. ` categories have the vertical alignment a bit off, you can notice the left side is not aligned with the right side, also the right side has an extra top border that shouldn't be there
![image](https://user-images.githubusercontent.com/3244654/82722693-4c426280-9cc9-11ea-8868-92cce8fd5401.png)
- [x] The `Subscribe`  & `News Letter` should look like the original
### `Subscribe` - Original:
![image](https://user-images.githubusercontent.com/3244654/82722750-b0652680-9cc9-11ea-9e49-69d4bab37526.png)
### `Newsletter` - Original:
![image](https://user-images.githubusercontent.com/3244654/82722760-c7a41400-9cc9-11ea-9502-973196906675.png)
- [x] The footer should look like the original
### Original:
![image](https://user-images.githubusercontent.com/3244654/82722783-01751a80-9cca-11ea-8775-499fd458e745.png)
### Small Screens (Smartphones) :iphone: 
- [x] The order of the top main content is wrong. While yours is `top, heroes of pandemic, culture, more stories, featured stories, the debate, opinion`, the original is `top stories, featured stories, the debate, opinion, more stories, heroes of pandemic, culture and travel`
- [x] The `Signup` feature keeps being a bit off
### Clone:
![image](https://user-images.githubusercontent.com/3244654/82722984-7dbc2d80-9ccb-11ea-92b8-91f019cfe754.png)
### Original:
![image](https://user-images.githubusercontent.com/3244654/82722987-8ca2e000-9ccb-11ea-9cae-af96877698f4.png)
- [x] The subsections of each dedicated category must be properly aligned along with the top content
![image](https://user-images.githubusercontent.com/3244654/82723047-06d36480-9ccc-11ea-8046-b62adae373de.png)
- [x] The `Subscribe`, `Newsletter` are still broken and are not looking like the original
### `Subscribe` - Original:
![image](https://user-images.githubusercontent.com/3244654/82723086-84977000-9ccc-11ea-8de2-ef845006f390.png)
### `Newsletter` - Original:
![image](https://user-images.githubusercontent.com/3244654/82723160-0a1b2000-9ccd-11ea-9ef4-aed813d260ea.png)

- [x] Footer needs more work as well, it should look like the original
![image](https://user-images.githubusercontent.com/3244654/82723136-de983580-9ccc-11ea-9989-eb251152a6ff.png)
### <em>Kindly fix the REQUIRED issues and send another code review request.

Cheers and Happy coding! :clap: 

Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of alexandru if you have any additional question)</em>
### :stop_sign: Status: Required Changes

simandebvu committed:
Removed attributes in CSS that are there in bootstrap

simandebvu committed:
Styled the navbar for desktops

simandebvu committed:
Styled the navbar for tablet and mobile

simandebvu committed:
Added row for editors picks

simandebvu committed:
Styled navbar responsive on all screen types

simandebvu committed:
Styled the Subscription and newsletter fo small medium & large screens

simandebvu committed:
Styled the newsletter

simandebvu committed:
Added featured articles

simandebvu committed:
Modified the start your day on medium screens

simandebvu committed:
Fixed the subscribe section on large screens

simandebvu committed:
Fixed navbar padding for small and large screens

simandebvu committed:
Added weather widget

simandebvu committed:
Fixed Stylelint errors

simandebvu committed:
Merge pull request #8 from simandebvu/feature-refactoring

Modified styles, layout and fonts

simandebvu commented:
## Changes made 

- Navbar with a weather widget and padding on large screens
- Changed font family and size for the navbar
- Subscribe, newsletter and news sections
- Modified footer with columns
- Added more content to the features section
- Removed code that is inherent in bootstrap
- Validated with W3c validator and stylelint
- Modified Signup feature for mobile
- Added image on the newsletter to show on large screens

santiago-dev96 commented:
# Pull Request under review by @santiago-rodrig (TSE)

santiago-dev96 reviewed (changes requested):
# Changes required :no_entry_sign:

You are getting closer, but there are some **details** that need to be resolved.

1. Each section must have a corresponding heading

<img width="262" alt="Screen Shot 2020-05-25 at 9 54 32 AM" src="https://user-images.githubusercontent.com/43162770/82819087-ce6b8c00-9e6d-11ea-91a2-3716590a321f.png">

2. The "More Stories" section needs a bit more separation between its items.

<img width="383" alt="Screen Shot 2020-05-25 at 9 54 41 AM" src="https://user-images.githubusercontent.com/43162770/82819172-eb07c400-9e6d-11ea-98e0-26810a8dfe0d.png">

3. The section with categories doesn't display each category as in the original page.

**Original**

<img width="1233" alt="Screen Shot 2020-05-25 at 9 57 17 AM" src="https://user-images.githubusercontent.com/43162770/82819387-4d60c480-9e6e-11ea-80f1-edd4e63f3ec9.png">

**Yours**

<img width="793" alt="Screen Shot 2020-05-25 at 9 57 08 AM" src="https://user-images.githubusercontent.com/43162770/82819427-5baee080-9e6e-11ea-96d9-ba77bd97af8c.png">

Please submit another CRR after implementing the changes.

You can find me in slack as **srodrig**, have a nice day! :wave:

simandebvu committed:
Added title for the sections

simandebvu committed:
Added spacing for opinion items

simandebvu committed:
CSS imporivements to list items

simandebvu committed:
W3c and stylelint validations

simandebvu committed:
Merge pull request #9 from simandebvu/feature-article-refactor

Feature article refactor

simandebvu commented:
## Changes done 😄 

- More stories and page list item improvements
- Added titles to sections and removed all uppercase titles so it looks more like the original
- Spacing between items in the __More__ section

addod19 commented:
## Pull Request under review by Daniel Larbi Addo (TSE)

addod19 reviewed (changes requested):
## Status: Require changes :recycle: :traffic_light: 

Hi @simandebvu, great work with this amazing project :+1: You did a great job fixing most of the comments pointed out by the previous TSE's :slightly_smiling_face: 
Let us address the comment below to make our project even better:
- The second image in the ```More stories``` is not displaying fully, kindly fix it
![mmore](https://user-images.githubusercontent.com/46388113/82846128-e88d8480-9ed6-11ea-91ec-455462a38a2f.png)

Kindly implement this change and submit another code review :smile:
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

simandebvu committed:
Removed scrolling effect

simandebvu committed:
Merge pull request #10 from simandebvu/feature-scroll-remove

Removed scrolling effect

simandebvu commented:
## Changes made 😄 

- Removed the scrolling effect from the __More Stories__ Section

christianotieno reviewed (changes requested):
# CHANGES REQUIRED :warning:
Hey @simandebvu   :wave:
-
You have done a splendid job! :tada: Keep it up.
We, however, do have some few changes to carry out before we proceed (see review comment).
##
## REQUIRED CHANGES
### LARGE SCREEN CHANGES: 
- [x] The general layout of the page is not properly centred as with the source page. The [`container`](https://www.w3schools.com/bootstrap4/bootstrap_containers.asp) class would be appropriate in the centring of the whole page content. See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|![newsweek](https://user-images.githubusercontent.com/22345961/82891790-05699c80-9f57-11ea-83be-b3b442686ef8.png)|![newsweek-clone](https://user-images.githubusercontent.com/22345961/82891816-11555e80-9f57-11ea-9b44-94609df209de.png)|
##
- [x] In the magazine section is not properly styled as with the source page.. See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|![in-the-magazine](https://user-images.githubusercontent.com/22345961/82892456-28488080-9f58-11ea-8ecc-28ec3f4dd7fa.png)|![in-the-magazine-clone](https://user-images.githubusercontent.com/22345961/82892479-33031580-9f58-11ea-84a0-14d8267d7793.png)|
##
- [x] Editor's pick section. See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|![editors-pick](https://user-images.githubusercontent.com/22345961/82892752-b3c21180-9f58-11ea-907e-04ed99c61ee5.png)|![editors-pick-clone](https://user-images.githubusercontent.com/22345961/82892781-c2a8c400-9f58-11ea-90a5-6da37edd00d7.png)|
##
- [x] `US`, `World`, `Business`, `Tech & Science`, and `Sports`'s section See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|![US-section](https://user-images.githubusercontent.com/22345961/82893527-cf79e780-9f59-11ea-994c-29ef20a2a488.png)|![US-section-clone](https://user-images.githubusercontent.com/22345961/82893618-f7694b00-9f59-11ea-9993-951c802062cb.png)|

- [x] The `footer` and the `subscribe` section. See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|![foooter](https://user-images.githubusercontent.com/22345961/82894344-1d431f80-9f5b-11ea-8da7-ae0fc7c8be87.png)|![foooter-clone](https://user-images.githubusercontent.com/22345961/82894436-3f3ca200-9f5b-11ea-8b55-7ad48d3f71e3.png)|

##
- [x] Comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)).

- [x] Make sure that basic CSS bad practices are not used ([article for support ](https://speckyboy.com/good-bad-css-practices)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) from Pathwright).

### MEDIUM & SMALL SCREEN CHANGES: 
-  [x] Conduct the same changes in the medium and small screen changes as well. 
- The goals here will be to get the clone to appear strikingly similar to the source page. Having assessed your skills I believe you are capable of cons=duciting these changes.  Happy coding.

 ## OPTIONAL
- [x] Validate your CSS using this [tool](https://jigsaw.w3.org/css-validator/validator).

- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

simandebvu committed:
Added container to page

simandebvu committed:
Adjusted images for editors picks and in the magazine

simandebvu committed:
Resolved footer in FULL HD Display

simandebvu committed:
Fixed Stylelint Errors

simandebvu committed:
Styled the newsletter for large to full HD Screens

simandebvu commented:
## Changes made

- Added removed container-fluid and added a container element to the body
- Edited the **in the magazine** and **editors picks** to look more like the original
- Resolved the **subscribe part** not rendering properly in Full HD Screens
- Newsletter image improvements
- CSS and HTML validation

simandebvu committed:
Removed unnecessary padding in wrapper

christianotieno reviewed (changes requested):
Hello @simandebvu 👋 
You have done a great job with the changes. Just some few more changes and we should be done here.
## Required
- During the screen breakpoints, the contents in both the medium and small screens should cover the whole width of the screen. The `container` class is only appropriate for large screens only. Kindly fix this issue then submit for another code review. Thank you

simandebvu committed:
Responsiveness-Added full screen for medium displays

simandebvu committed:
Stylelint resolved

simandebvu commented:
## Changed

- Fullscreen for medium and small screens

trekab commented on assets/css/style.css:
```diff
@@ -0,0 +1,375 @@
+  font-size: 14px;
+}
+
+@media (min-width: 768px) {
+  .custom-navbar-button {
+    margin-right: 0;
+  }
+
+  .news-wrapper {
+    overflow: hidden;
+    min-height: 270px;
+    max-height: 270px;
+  }
+
+  .subs-wrapper {
+    background-color: #f4f5f6;
+    overflow: hidden;
+    max-height: 270px;
+  }
+
+  .custom-navbar-button button {
+    font-size: 1rem;
+  }
+
+  .current-edition {
+    position: absolute;
+    top: 15px;
+    left: 90px;
+  }
+
+  .newsletter-thumb {
+    display: none;
+  }
+}
+
+@media screen and (min-width: 992px) {
+  .nav-button {
+    font-size: 17px;
+  }
```
This is a great place to rely on bootstrap classes. [Responsive breakpoints](https://getbootstrap.com/docs/4.3/layout/overview/#responsive-breakpoints)

trekab reviewed (changes requested):
# Changes Required! :recycle: 
Hi @simandebvu, you have done a good job implementing the feedback from the T.S.E's. However, there are a few changes to make.
- [x] For the `Featured slideshows` section, your clone looks different from the original on medium screens.
__Clone__:
![Clone](https://github.com/trekab/images/blob/master/img/shingi/clone.png?raw=true)
__Source__:
![Source](https://github.com/trekab/images/blob/master/img/shingi/source.png?raw=true)
- [x] Using Bootstrap Css classes **as mauch as possible** for styling (for example [Grid layout](https://getbootstrap.com/docs/4.3/layout/overview/) and [Flex](https://getbootstrap.com/docs/4.3/layout/utilities-for-layout/#flexbox-options)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/using_bootstrap/bootstrap_classes_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540308/)) is ideal for this project. Check the comment below for an example of where you could use bootstrap.

__Please make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__

simandebvu committed:
Responsiveness - Styled the featured slideshows to look like the original on medium screens

simandebvu committed:
Bootstrap- removed padding from css to using bootstrap, removed redundant classes

simandebvu committed:
Bootstrap -removed resdundant CSS code for nav button font

simandebvu committed:
Bootstrap -added bootstrap text-uppercase for subscribe

simandebvu commented:
## Changes made

- Updated featured slide show section on md devices
- Added more bootstrap classes

christianotieno reviewed (approved):
# PROJECT APPROVED :champagne:
Excellent job with the changes @simandebvu  :fireworks: 
##
***All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.***
- Congratulations on the splendid job.
- You can now merge to master. I wish you all the very best in your next tasks 🥇 
##
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
